specs/protocol/holocene/derivation.md

```diff
@@ -0,0 +1,186 @@
+
+## Steady Batch Derivation
+
+Steady Batch Derivation changes the derivation rules for invalid batches, and instead of giving them
+a second chance for a worst-case full sequencing window, they are immediately derived as
+deposit-only blocks. There are two stages at which this new rule takes effect, in the batch queue
+and in the engine queue.
+
+### Batch Queue
+
+With Holocene, whereever a singular batch would be dropped before, it is now immediately replaced
+by a deposit-only batch.
+
+TODO: details
+
+### Engine Queue
+
+Holocene changes the way in which the Engine Queue responds to payload attributes errors: On a
+payload processing error, the payload is immediately replaced by the respecive empty deposit-only
+attributes for the same block number and payload processing is reattempted with these new
+attributes. As before, a failure to process deposit-only attributes is a critical error.
+
+This new rule applies to both, processing of attributes to advance the safe or unsafe chain.
+This is in contract to just dropping the payload pre-Holocene and waiting for possible future valid
+payload attributes.
+
+## Partial Span Batch Validity
+
+Partial Span Batch Validity changes the span batch validation rules by only invalidating the
+remaining span batch once an invalid singular batch is encountered. Prior to this, span batch
+validity was treated atomically, so earlier pending valid singular batches could be invalidated by a
+later invalid singular batch pulled from a span batch.
+To accomplish this new behavior, the [Delta span batch batch queue rules](../delta/span-batches.md#batch-queue)
+are changed in the following ways.
+
+The max sequencer time drift check is applied to each singluar batch derived from a span batch
+individually. Once a singular batch is found to be invalid, only the remaining span batch is
+dropped. Note that the sequencing window check still applies to the span batch in full, because the
+oldest blocks would be the first to be invalidated and if dropped would also invalidate newer
+blocks.
+
+TODO: other changes to validation checks.
+
+## Strict Batch Ordering
+
+The new Strict Batch Ordering rule requires batches within and across channels to be strictly
+ordered. Note that this is already guaranteed for singular batcher derived from span batches by design.
+
+The strict ordering rules are implemented in the derivation pipeline by applying the following
+changes. 
+
+### Frame Queue
+
+The frame queue retains its function and holds all frames of the last batcher transaction. Holocene
+still allows multiple frames per batcher transaction. As before, this allows for optionally filling
+up remaining space of a batcher transactions with a starting frame of the next channel.
+
+### Channel Bank
+
+The Channel Bank becomes much simpler and only builds up a single channel.
+Channel frames have to arrive in order, and only one channel can be open at a time.
```

Comment:
Is it worth noting here that the current implementation shouldn't _need to_ change and supporting multiple channels is required for backwards-compatibility?

Reply:
The pre-Holocene derivation still has to be supported yes! I suggest consensus clients actually implement alternative stages pre and post holocene and then switch over at the L1 holocene activation block, including clearing the derivation pipeline state. This is also described in the new activation section.

So the legacy channel bank that supports buffering of multiple channels should be retained. However, after the FP challenge window expired, we could switch FP program implementations to a simpler DP implementation that only supports Holocene derivation, since FPs don't need to support historical derivation after a fork is active for long enough.